### PR TITLE
Bug 1666768 Truncate histogram values above 2^31

### PR DESCRIPTION
**Pull request by jklukas**
Reverts the temporary fix from #1333 and gets `internet_outages` working. I ran the `global_outages_v1` job locally and verified that it avoids the overflow issue with these changes.

### CODE REVIEW

File: mozfun/hist/mean/udf.sql [hunk 1]
```diff
@@ -19,7 +19,18 @@ CREATE OR REPLACE FUNCTION hist.mean(histogram ANY TYPE) AS (
   ELSE
     SAFE_CAST(
       TRUNC(
-        histogram.sum / (SELECT SUM(value) FROM UNNEST(histogram.values) WHERE value > 0)
+        histogram.sum / (
+          SELECT
+            SUM(
+              -- Truncate pathological values that are beyond the documented limits per
+              -- https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/collection/histograms.html#histogram-values
+              IF(value > 2147483648, 2147483648, value)
```
Comment (relud) [inline mozfun/hist/mean/udf.sql:27]:
```suggestion
              LEAST(2147483648, value)
```

File: sql/internet_outages/global_outages_v1/query.sql [hunk 2]
```diff
@@ -23,7 +23,9 @@ CREATE TEMP FUNCTION sum_values(x ARRAY<STRUCT<key INT64, value INT64>>) AS (
     ),
     b AS (
       SELECT
-        sum(value) AS t
+        -- Truncate pathological values that are beyond the documented limits per
+        -- https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/collection/histograms.html#histogram-values
+        SUM(IF(value > 2147483648, 2147483648, value)) AS t
```
Comment (relud) [inline sql/internet_outages/global_outages_v1/query.sql:28]:
```suggestion
        SUM(LEAST(2147483648, value)) AS t
```

File: sql/internet_outages/global_outages_v1/query.sql [hunk 3]
```diff
@@ -259,7 +261,7 @@ dns_success_time AS (
         client_id,
         time_slot,
         key,
-        sum(value) AS count
+        sum(IF(value > 2147483648, 2147483648, value)) AS count
```
Comment (relud) [inline sql/internet_outages/global_outages_v1/query.sql:264]:
```suggestion
        sum(LEAST(2147483648, value)) AS count
```

File: sql/internet_outages/global_outages_v1/query.sql [hunk 4]
```diff
@@ -316,7 +318,7 @@ dns_failure_src AS (
     client_id,
     time_slot,
     key,
-    sum(value) AS count
+    sum(IF(value > 2147483648, 2147483648, value)) AS count
```
Comment (relud) [inline sql/internet_outages/global_outages_v1/query.sql:321]:
```suggestion
    SUM(LEAST(2147483648, value)) AS count
```

File: sql/internet_outages/global_outages_v1/query.sql [hunk 5]
```diff
@@ -450,7 +452,7 @@ tls_handshake_time AS (
         client_id,
         time_slot,
         key,
-        sum(value) AS count
+        sum(IF(value > 2147483648, 2147483648, value)) AS count
```
Comment (relud) [inline sql/internet_outages/global_outages_v1/query.sql:455]:
```suggestion
        SUM(LEAST(2147483648, value)) AS count
```

**Review by relud — APPROVED** [review]
r+wc

**Review by Dexterp37 — APPROVED** [review]
Thanks for doing this.

**Pull request merged**